gio-subclass/src/action.rs

```diff
@@ -0,0 +1,220 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir @ 5b90945+)
+// from gir-files (https://github.com/gtk-rs/gir-files @ b215ee8+)
+// DO NOT EDIT
+
+#[allow(unused_imports)]
+use libc::{c_int, c_char, c_uchar, c_float, c_uint, c_double,
+    c_short, c_ushort, c_long, c_ulong,
+    c_void, size_t, ssize_t, intptr_t, uintptr_t, time_t, FILE};
+
+#[allow(unused_imports)]
+use glib::{gboolean, gconstpointer, gpointer, GType};
+
+use gobject_subclass::anyimpl::*;
+use gobject_subclass::object::*;
+
+#[cfg(any(feature = "v2_38", feature = "dox"))]
+use Error;
+use ffi;
+use glib;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+
+pub trait ActionImpl: AnyImpl + 'static {
+
+    fn activate(&self, action: &T, parameter: Option<&glib::Variant>);
```

Comment:
The `&T` here should be a `&gio::Action` AFAIU

Reply:
True. I was thinking for a moment that we could make actionImpl generic over T? But i'm not sure if that has any benefits to it

Reply:
That has the problem that you can't use `Box<ActionImpl>` anymore but have to provide the type parameter everywhere, or not?

Reply:
Not sure. I can't even remember why I wanted to make it generic

Reply:
Let's not do that for now then :) It can always be dynamically casted if needed